### ISSUE TITLE
What is this circle icon in the VCS section?

### ISSUE THREAD

**Issue by dharkness** [bhilburn/powerlevel9k#962]
There's a solid circle in the VCS section pictured below (right side of the left prompt) that I cannot find in all the icons with `get_icon_names`. Does anyone know what it is? It shows in all my git repos.

![image](https://user-images.githubusercontent.com/341319/44302502-d385c980-a2de-11e8-9ef3-651a45c6e155.png)

**Comment by onaforeignshore** [comment]
What is your font configuration?

**Comment by onaforeignshore** [comment]
So the circle appears for compatible/powerline fonts to show that you have unstaged changes in your working copy.

**Comment by dharkness** [comment]
Oh sorry. I'm using "DejaVuSansMono Nerd Font Mono Book". Shouldn't whatever I can see using `get_icon_names` look the same in the actual prompts? I've been able to change all the other [icons](https://nerdfonts.com/#cheat-sheet) so far.

**Comment by dharkness** [comment]
Is that icon hard-coded? You can see I have it set to a circle with a plus in it in the screenshot above.

    POWERLEVEL9K_VCS_UNSTAGED_ICON="\uf059"

**Comment by dharkness** [comment]
After playing around a bit more, it looks like that solid circle icon means the local repo is dirty, meaning it has untracked, unstaged, or staged changes. I've been able to display the untracked and staged icons, but not unstaged. I see where `vcs.zsh` sets `VCS_WORKDIR_DIRTY`, but not where the solid circle icon is added as a result.

**Comment by dharkness** [comment]
Yes, but here's my config:

    POWERLEVEL9K_VCS_UNSTAGED_ICON="\uF06A"

Echoing that Unicode character in the terminal displays a circle with an exclamation mark in it. Yet, the prompt shows a solid circle instead.

  POWERLEVEL9K_VCS_UNSTAGED_ICON="\uF06A"

I've changed this from the screenshot above which shows a circle with a plus inside it. Either way, I have _no_ icons set to a solid circle.

**Comment by dritter** [comment]
You need to set that variable before you source P9K. That should do the trick.

BUT: This seems not to be the real problem here. Looking at your screenshot, it looks like your MODE isn't set up correctly. I see that, because the branching icon should be a different one in `nerdfont-complete`, etc. The POWERLEVEL9K_MODE should be set before you source the theme as well. Could you show us your configuration?

**Comment by dharkness** [comment]
That seems to have done the trick. I had put all my P9K config at the end of my `.zshrc` with my aliases and such. I guess that works for colors and most icons, but not the theme. The untracked icon is now fixed. I was also able to remove some of the icon settings I had to override in my previous setup; they're now using the correct default icon, I assume due to the theme.

Thanks! I hadn't even considered that the theme wasn't working since it was mostly correct already.
